### ISSUE TITLE
Should be possible to avoid printing lifetimes when it only occurs once in a type

### ISSUE THREAD

**Issue by eriksvedang** [carp-lang/Carp#969]
Right?

Would make things a *lot* more readable...

**Comment by eriksvedang** [comment]
Also, we should probably make them stand out more visually.

**Comment by scolsen** [comment]
`type` currently prints lifetimes with angle brackets:

```
鲤 (type "foo")
=> (Ref String <a>)
```

Maybe we can adopt that everywhere?

Actually, it'd be good to adopt this in the parser for type sigs even, since it would help distinguish between type vars and lifetime vars (which would help w/ #910

**Comment by eriksvedang** [comment]
Yeah, some kind of signifier would be great. Perhaps angle brackets are OK, though they are a bit annoying to type – any other ideas?

**Comment by eriksvedang** [comment]
A sigil/prefix (like `&` or `@`) would be better imo.

**Comment by scolsen** [comment]
I think `&` makes a lot of sense since it's already a shorthand for refs!

**Comment by eriksvedang** [comment]
Oh, I actually meant *similar* to those. I think the signatures will be too hard to visually parse if we “overload” &, even though the compiler can discern them.

**Comment by scolsen** [comment]
Ahh gotcha, maybe `*` for nods to pointers?

**Comment by eriksvedang** [comment]
~t
.t

**Comment by scolsen** [comment]
`~` is nice too

**Comment by eriksvedang** [comment]
Yes that could work too.

**Comment by hellerve** [comment]
Don’t forget that `~` is already used for calling function references as well. That’s not a deal breaker in refs, but could be visually, confusing.

**Comment by TimDeve** [comment]
I really like `<a>` in term of readability but I understand that might make parsing harder.
